actions overview/concepts page (#24)
FlutterFlow/flutterflow-documentation: docs/resources/action-flow-editor/overview.md

---
slug: overview
title: Overview

> tags: []

Reviewer: Might be useful to add something about action output variables?

toc_max_heading_level: 5
---
Effectively managing user interactions is essential for developing interactive applications. Designing interactivity involves two steps:

1. Listening for Interaction (**Triggers**)
2. Responding to Interaction (**Actions**)

> Both Triggers and Actions are just functions. Triggers are functions executed when a user performs a specific interaction, while Actions are functions executed in response to the triggered event. Common triggers are:

Reviewer: ```suggestion
Triggers represent a specific event, while Actions are functions executed in response to the triggered event. Common triggers are:
```

I know triggers are technically callback functions, but I think that might be a bit more detail than is necessary?

- Tapping on a button (**On Tap**)
- Selecting an option from a dropdown list (**On Selected**)
- Loading a page (**On Page Load**)

Actions are the functions that are executed after the trigger has registered an event.

## Action Flow Editor

> The Action Flow Editor is a visual, node-based editor used to define the functions for Triggers and Actions. This editor simplifies the process of creating and managing business logic.

Reviewer: "define the functions for Triggers and Actions" --> "configure the functions that run in response to a trigger"?
